### ISSUE TITLE
[Bug]: Default values and partition keys arent carried over

### ISSUE THREAD

**Issue by filip-halt** [milvus-io/milvus#24639]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: Master  
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus-master
```

### Current Behavior

Im not sure if this feature is fully implemented but I do see test cases being merged in. I checked out the code of pymilvus and it seems that the default value is making it to the protobuf so I believe this is a milvus side issue. Here is an example of the code:

```python
connections.connect()

s = [
    FieldSchema('a', DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema('b', DataType.FLOAT_VECTOR, dim=3),
    FieldSchema('c', DataType.INT64, default_value = 3)
]

schema = CollectionSchema(s)

pprint(schema.to_dict())

c = Collection('d', schema)

pprint(c.describe())

c.insert([{"b": [1,2,3]}])

utility.drop_collection("d")
```

The printouts are:
```bash
{'auto_id': True,
 'description': '',
 'fields': [{'auto_id': True,
             'description': '',
             'is_primary': True,
             'name': 'a',
             'type': <DataType.INT64: 5>},
            {'description': '',
             'name': 'b',
             'params': {'dim': 3},
             'type': <DataType.FLOAT_VECTOR: 101>},
            {'default_value': long_data: 3
,
             'description': '',
             'name': 'c',
             'type': <DataType.INT64: 5>}]}
```

which has the correct default set, but the collection when retrieved is missing the default: 

```bash
{'aliases': [],
 'auto_id': False,
 'collection_id': 441903835571032981,
 'collection_name': 'd',
 'consistency_level': 2,
 'description': '',
 'enable_dynamic_field': False,
 'fields': [{'auto_id': True,
             'default_value': None,
             'description': '',
             'field_id': 100,
             'is_dynamic': False,
             'is_partition_key': False,
             'is_primary': True,
             'name': 'a',
             'params': {},
             'type': 5},
            {'auto_id': False,
             'default_value': None,
             'description': '',
             'field_id': 101,
             'is_dynamic': False,
             'is_partition_key': False,
             'is_primary': False,
             'name': 'b',
             'params': {'dim': 3},
             'type': 101},
            {'auto_id': False,
             'default_value': None,
             'description': '',
             'field_id': 102,
             'is_dynamic': False,
             'is_partition_key': False,
             'is_primary': False,
             'name': 'c',
             'params': {},
             'type': 5}],
 'num_partitions': 1,
 'num_shards': 1,
 'properties': []}
```

I double checked the schema was correctly being converted to protobuff and it is:

```python
req = milvus_types.CreateCollectionRequest(collection_name=collection_name,
                                                   schema=bytes(schema.SerializeToString()),
                                                   consistency_level=consistency_level)
        
        print(schema_types.CollectionSchema.FromString(req.schema))
```
prints:

```bash
name: "d"
autoID: true
fields {
  name: "a"
  is_primary_key: true
  data_type: Int64
  autoID: true
}
fields {
  name: "b"
  data_type: FloatVector
  type_params {
    key: "dim"
    value: "3"
  }
}
fields {
  name: "c"
  data_type: Int64
  default_value {
    long_data: 3
  }
}
```

### Expected Behavior

For the default to be used when not passing in value, currently it errors out: 
`pymilvus.exceptions.ParamError: <ParamError: (code=1, message=Field c don't match in entities[0])>`

**Comment by yanliang567** [comment]
/assign @NicoYuan1986 
could you please help to reproduce the issue?
/unassign

**Comment by filip-halt** [comment]
in addition to default, partition key is also not working.

**Comment by yanliang567** [comment]
> in addition to default, partition key is also not working.

the pr of partition key was not merged into master yet.

**Comment by xiaofan-luan** [comment]
already merged into master

**Comment by NicoYuan1986** [comment]
For the ParamError:
we don't support default value on json field and insert array type.
To use default value, we suggest insert list of list or DataFrame.

examples:
```python
# 1. list of list
data = [
    [[1, 2, 3]]
]
c.insert(data)

# 2. DataFrame
data = pd.DataFrame({
    "b": [[1, 2, 3]]
})
c.insert(data)
```



```
c.insert([{"b": [1,2,3]}])
```
This array will report error. 
```
c.insert([{"b": [1, 2, 3], "c": 1}])
```
This will be ok.

I think we can add more documentation. @smellthemoon

**Comment by NicoYuan1986** [comment]
For c.describe(),

we have a case test_collection_describe for test. 
There is an issue #24493 and a fix pr milvus-io/pymilvus#1488 for it.
I think it is fixed now.
